adapters/coap-vertx/src/main/java/org/eclipse/hono/adapter/coap/vertx/VertxBasedCoapAdapter.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.adapter.coap.vertx;
+
+import org.eclipse.californium.core.CoapServer;
+import org.eclipse.californium.core.coap.CoAP.ResponseCode;
+import org.eclipse.californium.core.server.resources.CoapExchange;
+import org.eclipse.hono.adapter.coap.AbstractVertxBasedCoapAdapter;
+import org.eclipse.hono.adapter.coap.CoapAdapterProperties;
+import org.eclipse.hono.adapter.coap.CoapCredentialsStore;
+import org.eclipse.hono.adapter.coap.ExtendedDevice;
+import org.eclipse.hono.util.Constants;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+
+/**
+ * A Vert.x based Hono protocol adapter for accessing Hono's Telemetry &amp; Event API using COAP.
+ */
+public final class VertxBasedCoapAdapter extends AbstractVertxBasedCoapAdapter<CoapAdapterProperties> {
+
+    private static final String TELEMETRY_RESOURCE = "telemetry";
+    private static final String EVENT_RESOURCE = "event";
```

Comment:
OK, though I was starting using the http-adapter, I got mislead, because the URI parts there are not based on those definitions. e.g.:

`router.route(HttpMethod.PUT, "/telemetry/*").handler(basicAuthHandler);`

So there is a plan, to consequently use those defined strings?